src/site/content/en/blog/predictive-prefetching/index.md

```diff
@@ -0,0 +1,128 @@
+
+## Prefetching JavaScript
+
+![Prefetching example](guess-1.png)
+
+Prefetching is a technique which allows you to download in the background and save into the cache resources that you think the user will need soon. The usual way to do this is by using `<link rel="prefetch">`. Two common pitfalls with prefetching are:
+
+- Prefetching too many resources and consuming a lot of data (*overfetching*)
+- Never prefetching the resources the user needs
+
+Predictive allows you to solve these problems by consuming a report for the users' navigational patterns.
+
+## Predictive Prefetching with Guess.js
+
+A JavaScript library which provides predictive prefetching functionality is [Guess.js](https://github.com/guess-js). Guess.js consumes a report from Google Analytics or another analytics provider to build a predictive model that could be used for smartly prefetching only what the user may need in the future.
+
+Guess.js has integration with Angular, Next.js, Nuxt.js, and Gatsby. To use it in your application, you should add a single line of webpack configuration specifying a [Google Analytics view ID](https://stackoverflow.com/questions/36898103/what-is-a-viewid-in-google-analytics):
+
+```js/0,5
+const { GuessPlugin } = require('guess-webpack');
+
+// ...
+plugins: [
+   // ...
+   new GuessPlugin({ GA: 'XXXXXX' })
+]
+// ...
+```
+
+If you're not using Google Analytics, you can specify a `reportProvider` and download data from your favorite service.
+
+### Integration with Angular
+
+To integrate Guess.js with an Angular CLI project, you can use [`ngx-build-plus`](https://github.com/manfredsteyer/ngx-build-plus). The steps you need to go through include:
+
+* Install of `ngx-build-plus`, `guess-webpack`, and `guess-parser`
+* Add the `GuessPlugin` as part of your webpack configuration specifying a Google Analytics view ID or, if using different analytics provider, a `reportProvider`:
+
+```js
+const { GuessPlugin } = require('guess-webpack');
+const { parseRoutes } = require('guess-parser');
+
+module.exports = {
+  plugins: [
+    new GuessPlugin({
+      GA: 'VIEW_ID',
+      runtime: {
+	      delegate: false
+      },
+      routeProvider() {
+	      return parseRoutes('.');
+      }
+    })
+  ]
+};
+```
+
+* Build your application specifying your extended webpack config with the `--extra-webpack-config` option
+
+For a video introduction, check the video below.
+
+<div style="width:100%; padding-top: 56.25%; position: relative;">
+<iframe style="width:100%; height: 100%;position: absolute; top: 50%; left:
+50%; transform: translate(-50%,-50%);"
+src="https://www.youtube.com/embed/5FRxQiGqqmM" frameborder="0"
+allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
+allowfullscreen></iframe>
+</div>
+
+## How Does Guess.js Work
+
+Guess.js will add predictive prefetching going through the following steps:
```

Comment:
```suggestion
Here's how Guess.js implements predictive prefetching:
```
Suggest integrating this chunk into the "Predictive prefetching with Guess.js" section, right under the first paragraph. That way, the sequence goes:
1. What is Guess?
1. How does it work?
1. How can you use it? (via various integrations)